crates/bevy_winit/src/lib.rs

```diff
@@ -218,33 +154,13 @@ impl Plugin for WinitPlugin {
     }
 }
 
-#[derive(SystemParam)]
-struct WindowAndInputEventWriters<'w> {
-    // `winit` `WindowEvent`s
-    window_resized: EventWriter<'w, WindowResized>,
-    window_close_requested: EventWriter<'w, WindowCloseRequested>,
-    window_scale_factor_changed: EventWriter<'w, WindowScaleFactorChanged>,
-    window_backend_scale_factor_changed: EventWriter<'w, WindowBackendScaleFactorChanged>,
-    window_focused: EventWriter<'w, WindowFocused>,
-    window_occluded: EventWriter<'w, WindowOccluded>,
-    window_moved: EventWriter<'w, WindowMoved>,
-    window_theme_changed: EventWriter<'w, WindowThemeChanged>,
-    window_destroyed: EventWriter<'w, WindowDestroyed>,
-    lifetime: EventWriter<'w, ApplicationLifetime>,
-    keyboard_input: EventWriter<'w, KeyboardInput>,
-    character_input: EventWriter<'w, ReceivedCharacter>,
-    mouse_button_input: EventWriter<'w, MouseButtonInput>,
-    touchpad_magnify_input: EventWriter<'w, TouchpadMagnify>,
-    touchpad_rotate_input: EventWriter<'w, TouchpadRotate>,
-    mouse_wheel_input: EventWriter<'w, MouseWheel>,
-    touch_input: EventWriter<'w, TouchInput>,
-    ime_input: EventWriter<'w, Ime>,
-    file_drag_and_drop: EventWriter<'w, FileDragAndDrop>,
-    cursor_moved: EventWriter<'w, CursorMoved>,
-    cursor_entered: EventWriter<'w, CursorEntered>,
-    cursor_left: EventWriter<'w, CursorLeft>,
-    // `winit` `DeviceEvent`s
-    mouse_motion: EventWriter<'w, MouseMotion>,
+trait AppSendEvent {
+    fn send_event<E: bevy_ecs::event::Event>(&mut self, event: E);
+}
+impl AppSendEvent for App {
+    fn send_event<E: bevy_ecs::event::Event>(&mut self, event: E) {
+        self.world.send_event(event);
+    }
 }
```

Comment:
I like it, and I wouldn't block on it, but this might be better as a usability method within bevy_ecs maybe?

Reply:
I tried to add it earlier and it was controversial IIRC 😅 Private helper is fine though.